src/Avalonia.Controls.DataGrid/Collections/DataGridCollectionView.cs

```diff
@@ -3424,11 +3423,17 @@ private void ProcessCollectionChanged(NotifyCollectionChangedEventArgs args)
             }
 
             // fire notifications for adds
-            if ((args.Action == NotifyCollectionChangedAction.Add ||
-                args.Action == NotifyCollectionChangedAction.Replace) &&
-                (Filter == null || PassesFilter(addedItem)))
+            if (args.NewItems != null && 
+                (args.Action == NotifyCollectionChangedAction.Add ||
+                 args.Action == NotifyCollectionChangedAction.Replace))
```

Comment:
Seems replace still won't work well when multiple items are replaced at the same time. 
Should be similar code be added above for ProcessRemoveEvent?

Reply:
I'll have a look later, I really was only looking to fix adding multiple items but you are likely right about that. I didn't know whether or not replacing had issues as I hadn't tried.

Reply:
I just pushed a new commit which adds similar changes to removing items and that has also fixed removing multiple items, I wasn't too sure how to test replacing multiple items. In the process I did unintentionally cause a merge commit for my PR by clicking a sync branch button on github out of curiosity, and I know that rebasing is preferred and I'm not sure if that is any issue or not...

Reply:
No, it's not an issue.
Thanks! I will check later today or tomorrow.